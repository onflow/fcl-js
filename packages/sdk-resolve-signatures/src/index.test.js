import {resolveSignatures} from "./index.js"

test("exports function", () => {
  expect(typeof resolveSignatures).toBe("function")
})

test("Golden Path", async () => {
  const signingFunction = jest.fn(() => ({
    addr: "foo",
    keyId: 0,
    signature: "SIGNATURE",
  }))

  const ix = await resolveSignatures({
    tag: 4,
    message: {
      cadence: "foo",
      arguments: [],
      computeLimit: 10,
    },
    accounts: {
      foo: {
        addr: "foo",
        keyId: 0,
        sequenceNum: 0,
        signingFunction: signingFunction,
      },
    },
    proposer: "foo",
    authorizations: ["foo"],
    payer: "foo",
  })

  expect(ix).toMatchSnapshot()
})
