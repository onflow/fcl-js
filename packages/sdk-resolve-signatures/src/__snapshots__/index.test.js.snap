// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Golden Path 1`] = `
Object {
  "accounts": Object {
    "foo": Object {
      "addr": "foo",
      "keyId": 0,
      "sequenceNum": 0,
      "signature": "SIGNATURE",
      "signingFunction": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "addr": "foo",
              "interaction": [Circular],
              "keyId": 0,
              "message": "f845f84283666f6fc0a000000000000000000000000000000000000000000000000000000000000000000a870000000000000f8080870000000000000fc8870000000000000fc0",
              "roles": undefined,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Object {
              "addr": "foo",
              "address": "foo",
              "cid": "foo",
              "keyId": 0,
              "sig": "SIGNATURE",
              "signature": "SIGNATURE",
            },
          },
        ],
      },
    },
  },
  "authorizations": Array [
    "foo",
  ],
  "message": Object {
    "arguments": Array [],
    "cadence": "foo",
    "computeLimit": 10,
  },
  "payer": "foo",
  "proposer": "foo",
  "tag": 4,
}
`;
